fix: backup and restore tool

6qfxa opened the pull request:
The existing backup tool returns errors:
```
Fetching page 0...
Status endpoint returned an invalid status.
Invalid JSON content
```

It could be caused by the old version of ic-agent. Updating cargo dependencies is not easy due to breaking changes.

It is easier to rewrite the backup tool to directly call the canister backup/restore methods with the new version ic-agent.

The rewrite removes 650 lines of code, adds 170 lines of code (not counting Cargo.lock changes)

The existing LICENSE file is kept intact, but could be removed in favor of the top-level LICENSE
because the new code does not use code from `qu`.

chmllr commented on src/backend/env/mod.rs:
```diff
@@ -2594,7 +2594,7 @@ pub(crate) mod tests {
             0,
             None,
-            None,
+            Some("SYNAPSE".into()),
```
Synapse name in tests? 😅

taggrx replied:
This was implemented when they decided to use Taggr :)

Commit:
fix: backup and restore tool

The existing backup tool returns errors:
```
Fetching page 0...
Status endpoint returned an invalid status.
Invalid JSON content
```

It could be caused by the old version of ic-agent.
Updating cargo dependencies is not easy due to breaking changes.

It is easier to rewrite the backup tool to directly call the canister
backup/restore methods with the new version ic-agent.

The rewrite removes 650 lines of code, adds 170 lines of code
(not counting Cargo.lock changes)

xqxpx reviewed (approved):
Awesome patch! Thanks, @6qfxa and welcome to anonymity!
